task10/script.js

```diff
@@ -0,0 +1,516 @@
+let homeString = "<div id=\"userInfo\">\n" +
+    "        <h3 id=\"userInfoName\"></h3>\n" +
+    "        <img id=\"userInfoAvatar\"\n" +
+    "             alt=\"user avatar\">\n" +
+    "        <h3 class=\"userInfoH3\">Posts: 3</h3>\n" +
+    "        <h3 class=\"userInfoH3\">Likes: 3</h3>\n" +
+    "        <h3 class=\"userInfoH3\" id=\"favouriteHashtags\">Favourite hashtags:</h3>\n" +
+    "        <p>\n" +
+    "        <div class=\"tag\">#GoldenCup</div>\n" +
+    "        <div class=\"tag\">#WAT3-0LIV</div>\n" +
+    "        <div class=\"tag\">#Unbeaten</div>\n" +
+    "        <div class=\"tag\">#Arsenal</div>\n" +
+    "        <div class=\"tag\">#EPL</div>\n" +
+    "        </p>\n" +
+    "\n" +
+    "    </div>\n" +
+    "    <div id=\"posts\">\n" +
+    "        <div id=\"addTenDiv\">\n" +
+    "            <button class=\"button\" id=\"addTen\" onclick=\"addTen()\">Add 10 more</button>\n" +
+    "        </div>\n" +
+    "    </div>\n" +
+    "    <div id=\"search\">\n" +
+    "        <form>\n" +
+    "            <h3 id=\"filters\">Filters</h3>\n" +
+    "            <h3 class=\"searchH\">Author:</h3>\n" +
+    "            <input type=\"text\" id=\"author\">\n" +
+    "            <h3 class=\"searchH\">Hashtag:</h3>\n" +
+    "            <input type=\"text\" placeholder=\"#Hashtag\">\n" +
+    "            <h3 class=\"searchH\">From:</h3>\n" +
+    "            <input type=\"datetime-local\" placeholder=\"From\" id=\"from\">\n" +
+    "            <h3 class=\"searchH\">To:</h3>\n" +
+    "            <input type=\"datetime-local\" placeholder=\"To\" id=\"to\">\n" +
+    "            <button class=\"button\" id=\"searchButton\" onclick=\"searchFilter()\">Search</button>\n" +
```

Comment:
Filtering should be done not on button click, but on form's submit action. The same about login